src/Controller/DownloadController.php

```diff
@@ -111,6 +111,9 @@ public function httpAction(Dataset $dataset, Datastore $dataStore, LogActionItem
         if ($dataset->isRestricted()) {
             throw new BadRequestHttpException('Unable to download restricted dataset');
         }
+        if ($dataset->getDatasetStatus() !== Dataset::DATASET_STATUS_ACCEPTED) {
```

Comment:
You could combine the above condition with restricted and make a single `if` block with availability status